lib/roby/app.rb

```diff
@@ -1041,6 +1047,44 @@ def base_setup
             @base_setup_done = true
         end
 
+        def lock_log_dir
+            return if log_dir_locked?
+
+            return unless @lock_file
+
+            @lock_file ||= File.open(
+                File.join(@log_dir, LOCK_FILE_EXT),
+                File::RDWR | File::CREAT, 0o644
+            )
+            @lock_file.flock(File::LOCK_EX | File::LOCK_NB)
```

Comment:
There's a race condition here. The file is present by not yet locked.

When creating it, it would be better to create it with a different name, lock it, and then rename the file. This will allow the other tooling (e.g. log archive) to ignore the folder if it does not have a .lock file at all.